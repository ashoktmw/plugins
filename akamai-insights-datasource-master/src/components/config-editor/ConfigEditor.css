.credentials__container {
  height: 200px;
  width: 640px;
}

.credentials__textarea {
  height: 150px;
}
