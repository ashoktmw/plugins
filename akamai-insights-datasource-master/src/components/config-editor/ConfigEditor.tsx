import { Field, InlineField, Input, TextArea } from '@grafana/ui';
import { camelCase, isEmpty } from 'lodash';
import React, { ChangeEvent, useState } from 'react';

import { inputWidth, labelWidth, Secret, secretsNames } from './types';
import { DataSourceProps } from '../../types/types';
import './ConfigEditor.css';

export function ConfigEditor({ options, onOptionsChange }: DataSourceProps) {

  const [ credentialsTextAreaInvalid, setCredentialsTextAreaInvalid ] = useState<boolean>(false);
  const onInputChange = ({ target: { value } }: ChangeEvent<HTMLInputElement>, fieldName: string) => {
    const jsonData = {
      ...options.jsonData,
      [ fieldName ]: value
    };
    onOptionsChange({ ...options, jsonData });
  };

  const onClientSecretChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.ClientSecret);
  const onHostChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.Host);
  const onAccessTokenChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.AccessToken);
  const onClientTokenChange = (event: ChangeEvent<HTMLInputElement>) => onInputChange(event, Secret.ClientToken);
  const onCredentialsTextAreaChange = ({ target: { value } }: ChangeEvent<HTMLTextAreaElement>) => {
    const secrets = value
      .split('\n')
      .map(line => line.trim().split(/\s=\s(.*)/s))
      .map(([ name, secret ]) => ({ [ camelCase(name) ]: secret }))
      .reduce((prev, next) => ({ ...prev, ...next }), {});

    const entries = Object.entries(secrets);
    const isCredentialsTextAreaInvalid = entries.length !== secretsNames.length || entries
      .some(([ name, secret ]) => !secretsNames.includes(name as Secret) || isEmpty(secret));
    setCredentialsTextAreaInvalid(isCredentialsTextAreaInvalid);

    if (!isCredentialsTextAreaInvalid) {
      const jsonData = {
        ...options.jsonData,
        ...secrets
      };

      onOptionsChange({ ...options, jsonData });
    }
  };

  const { jsonData: { clientSecret, host, clientToken, accessToken } } = options;

  return (
    <div className="gf-form-group">
      <div className="credentials__container">
        <Field label="Information" description="Paste credentials here to auto fill form below">
          <TextArea
            className="credentials__textarea"
            name="credentialsTextArea"
            invalid={credentialsTextAreaInvalid}
            onChange={onCredentialsTextAreaChange}/>
        </Field>
      </div>
      <hr/>
      <InlineField
        label="Akamai Client Secret"
        labelWidth={labelWidth}>
        <Input
          onChange={onClientSecretChange}
          value={clientSecret || ''}
          placeholder="Enter client secret"
          width={inputWidth}
        />
      </InlineField>
      <InlineField
        label="Host"
        labelWidth={labelWidth}>
        <Input
          value={host || ''}
          placeholder="Enter host"
          width={inputWidth}
          onChange={onHostChange}
        />
      </InlineField>
      <InlineField
        label="Access Token"
        labelWidth={labelWidth}>
        <Input
          value={accessToken || ''}
          placeholder="Enter access token"
          width={inputWidth}
          onChange={onAccessTokenChange}
        />
      </InlineField>
      <InlineField
        label="Client Token"
        labelWidth={labelWidth}>
        <Input
          value={clientToken || ''}
          placeholder="Enter client token"
          width={inputWidth}
          onChange={onClientTokenChange}
        />
      </InlineField>
    </div>
  );
}
