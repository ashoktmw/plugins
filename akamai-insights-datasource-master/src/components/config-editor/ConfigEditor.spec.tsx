import { fireEvent, render } from '@testing-library/react';
import React from 'react';

import { ConfigEditor } from './ConfigEditor';

describe('ConfigEditor component', () => {
  it('should trigger onOptionsChange when one of inputs has changed', () => {
    const options: any = {
      jsonData: {
        clientToken: 'testToken'
      }
    };
    const onOptionsChange = jest.fn();

    const { getByPlaceholderText } = render(
      <ConfigEditor options={options} onOptionsChange={onOptionsChange}/>
    );

    fireEvent.change(getByPlaceholderText('Enter host'), { target: { value: 'akamai.com' } });

    expect(onOptionsChange).toHaveBeenCalledWith({
      jsonData: {
        clientToken: 'testToken',
        host: 'akamai.com'
      }
    });
  });

  it('should update all fields when credentials are pasted into textarea', () => {
    const options: any = {
      jsonData: {}
    };
    const onOptionsChange = jest.fn();

    const { container } = render(
      <ConfigEditor options={options} onOptionsChange={onOptionsChange}/>
    );

    fireEvent.change(container.querySelector('[name="credentialsTextArea"]') as Element, {
      target: {
        value: `client_secret = test=
                host = test.net
                access_token = test-123
                client_token = test-test`
      }
    });

    expect(onOptionsChange).toHaveBeenCalledWith({
      jsonData: {
        clientSecret: 'test=',
        host: 'test.net',
        accessToken: 'test-123',
        clientToken: 'test-test'
      }
    });
  });

  it('should mark textarea as invalid and not update form when credentials are in wrong format', () => {
    const options: any = {
      jsonData: {}
    };
    const onOptionsChange = jest.fn();

    const tree = render(
      <ConfigEditor options={options} onOptionsChange={onOptionsChange}/>
    );
    const textArea = tree.container.querySelector('[name="credentialsTextArea"]') as Element;

    fireEvent.change(textArea, {
      target: {
        value: `clientsecret = test=
                host = test.net
                accesstoken = test-123
                client_token = test-test`
      }
    });

    expect(onOptionsChange).not.toHaveBeenCalled();
    expect(tree).toMatchSnapshot();
  });
});
