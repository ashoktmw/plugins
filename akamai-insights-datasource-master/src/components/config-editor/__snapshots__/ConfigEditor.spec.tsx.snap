// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigEditor component should mark textarea as invalid and not update form when credentials are in wrong format 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="gf-form-group"
      >
        <div
          class="credentials__container"
        >
          <div
            class="css-8e5b3"
          >
            <div
              class="css-jt4xma-Label"
            >
              <label>
                <div
                  class="css-xhqy0o"
                >
                  Information
                </div>
                <span
                  class="css-zjkzrp-Label-description"
                >
                  Paste credentials here to auto fill form below
                </span>
              </label>
            </div>
            <div>
              <div>
                <textarea
                  class="css-b6cogb credentials__textarea"
                  name="credentialsTextArea"
                />
              </div>
            </div>
          </div>
        </div>
        <hr />
        <div
          class="css-1f5gueu"
        >
          <label
            class="css-tomx28"
          >
            Akamai Client Secret
          </label>
          <div
            class="css-1rfunm5"
          >
            <div
              class="css-yy0rti-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-1w5c5dq-input-inputWrapper"
              >
                <input
                  class="css-1n6y0bv-input-input"
                  placeholder="Enter client secret"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1f5gueu"
        >
          <label
            class="css-tomx28"
          >
            Host
          </label>
          <div
            class="css-1rfunm5"
          >
            <div
              class="css-yy0rti-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-1w5c5dq-input-inputWrapper"
              >
                <input
                  class="css-1n6y0bv-input-input"
                  placeholder="Enter host"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1f5gueu"
        >
          <label
            class="css-tomx28"
          >
            Access Token
          </label>
          <div
            class="css-1rfunm5"
          >
            <div
              class="css-yy0rti-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-1w5c5dq-input-inputWrapper"
              >
                <input
                  class="css-1n6y0bv-input-input"
                  placeholder="Enter access token"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="css-1f5gueu"
        >
          <label
            class="css-tomx28"
          >
            Client Token
          </label>
          <div
            class="css-1rfunm5"
          >
            <div
              class="css-yy0rti-input-wrapper"
              data-testid="input-wrapper"
            >
              <div
                class="css-1w5c5dq-input-inputWrapper"
              >
                <input
                  class="css-1n6y0bv-input-input"
                  placeholder="Enter client token"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="gf-form-group"
    >
      <div
        class="credentials__container"
      >
        <div
          class="css-8e5b3"
        >
          <div
            class="css-jt4xma-Label"
          >
            <label>
              <div
                class="css-xhqy0o"
              >
                Information
              </div>
              <span
                class="css-zjkzrp-Label-description"
              >
                Paste credentials here to auto fill form below
              </span>
            </label>
          </div>
          <div>
            <div>
              <textarea
                class="css-b6cogb credentials__textarea"
                name="credentialsTextArea"
              />
            </div>
          </div>
        </div>
      </div>
      <hr />
      <div
        class="css-1f5gueu"
      >
        <label
          class="css-tomx28"
        >
          Akamai Client Secret
        </label>
        <div
          class="css-1rfunm5"
        >
          <div
            class="css-yy0rti-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-1w5c5dq-input-inputWrapper"
            >
              <input
                class="css-1n6y0bv-input-input"
                placeholder="Enter client secret"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1f5gueu"
      >
        <label
          class="css-tomx28"
        >
          Host
        </label>
        <div
          class="css-1rfunm5"
        >
          <div
            class="css-yy0rti-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-1w5c5dq-input-inputWrapper"
            >
              <input
                class="css-1n6y0bv-input-input"
                placeholder="Enter host"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1f5gueu"
      >
        <label
          class="css-tomx28"
        >
          Access Token
        </label>
        <div
          class="css-1rfunm5"
        >
          <div
            class="css-yy0rti-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-1w5c5dq-input-inputWrapper"
            >
              <input
                class="css-1n6y0bv-input-input"
                placeholder="Enter access token"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-1f5gueu"
      >
        <label
          class="css-tomx28"
        >
          Client Token
        </label>
        <div
          class="css-1rfunm5"
        >
          <div
            class="css-yy0rti-input-wrapper"
            data-testid="input-wrapper"
          >
            <div
              class="css-1w5c5dq-input-inputWrapper"
            >
              <input
                class="css-1n6y0bv-input-input"
                placeholder="Enter client token"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
